src/containers/Assign/Reassignment.tsx

```diff
@@ -0,0 +1,80 @@
+import React, { useState } from 'react'
+import { Label, Grid } from 'semantic-ui-react'
+import { useLanguageProvider } from '../../contexts/Localisation'
+import { AssignmentOption } from '../../utils/data/assignmentOptions'
+import useReasignReviewAssignment from '../../utils/hooks/useReassignReviewAssignment'
+import { AssignmentDetails, PageElement } from '../../utils/types'
+import AssigneeDropdown from './AssigneeDropdown'
+import useGetAssignmentOptions from './useGetAssignmentOptions'
+
+interface ReassignmentProps {
+  assignments: AssignmentDetails[]
+  sectionCode: string
+  elements: PageElement[]
+  isLastLevel: (selectedIndex: number) => boolean
+  shouldAssignState: [number | boolean, React.Dispatch<React.SetStateAction<number | boolean>>]
+  previousAssignee: number
+}
+
+const Reassignment: React.FC<ReassignmentProps> = ({
+  assignments,
+  sectionCode,
+  elements,
+  isLastLevel,
+  shouldAssignState,
+  previousAssignee,
+}) => {
+  const { strings } = useLanguageProvider()
+  const getAssignmentOptions = useGetAssignmentOptions()
+  const [reassignmentError, setReassignmentError] = useState(false)
+  const { reassignSection } = useReasignReviewAssignment()
+
+  const assignmentOptions = getAssignmentOptions(
+    {
+      assignments,
+      sectionCode,
+      elements,
+      previousAssignee,
+    },
+    null
+  )
+  if (!assignmentOptions) return null
+
+  const onReassignment = async (value: number) => {
+    if (value === assignmentOptions.selected) return console.log('Re-assignment to same reviewer')
+    if (value === AssignmentOption.UNASSIGN) return console.log('un assignment not implemented')
+
+    const previousAssigmnet = assignments.find(
+      (assignment) => assignment.reviewer.id === previousAssignee
+    )
+    const reassignment = assignments.find((assignment) => assignment.reviewer.id === value)
+
+    if (!reassignment) return
+    try {
+      await reassignSection({
+        unassignmentId: previousAssigmnet?.id,
+        reassignment,
+        sectionCode,
+        elements,
+      })
+    } catch (e) {
+      console.log(e)
```

Comment:
```suggestion
      console.error(e)
```
I like using this for errors these days, but it's not a biggie.